src/components/fixes/PatchedNextLink.tsx

```diff
@@ -0,0 +1,23 @@
+import Link, { LinkProps } from 'next/link'
+import { ForwardRefRenderFunction, forwardRef } from 'react'
+
+type PatchedNextLinkProps = LinkProps & {
+  children: React.ReactNode
+  className?: string
+}
+
+// https://headlessui.com/react/menu#integrating-with-next-js
+const PatchedNextLink: ForwardRefRenderFunction<
```

Comment:
**question** what you think about calling this JuiceLink or similar?

We could use this in place of `next/link` pretty much everywhere 🤔

Reply:
I think same as above comment, lets leave this as a specific fix for integration with headless ui (which is passing through a ref)